.ci/scripts/test.sh

```diff
@@ -119,39 +117,101 @@ if [ "${interface}" == "daal/java" ]; then
 fi
 
 for link_mode in ${link_modes}; do
-    # Release Examples testing
     if [ "${link_mode}" == "static" ]; then
+        lib_ext="a"
         l="lib"
     elif [ "${link_mode}" == "dynamic" ]; then
+        lib_ext="so"
         if [ "${OS}" == "lnx" ]; then
             l="so"
         else
             l="dylib"
         fi
     fi
-    build_command="make ${make_op} ${l}${full_arch} mode=build compiler=${compiler}"
-    echo "Building ${TEST_KIND} ${build_command}"
-    (${build_command})
-    err=$?
-    if [ ${err} -ne 0 ]; then
-        echo -e "$(date +'%H:%M:%S') BUILD FAILED\t\t${link_mode}"
-        TESTING_RETURN=${err}
-        continue
-    else
-        echo -e "$(date +'%H:%M:%S') BUILD COMPLETED\t\t${link_mode}"
-    fi
-    run_command="make ${l}${full_arch} mode=run compiler=${compiler}"
-    echo "Running ${TEST_KIND} ${run_command}"
-    (${run_command})
-    err=$?
-    if [ ${err} -ne 0 ]; then
-        echo -e "$(date +'%H:%M:%S') RUN FAILED\t\t${link_mode} with errno ${err}"
-        TESTING_RETURN=${err}
-        continue
+    if [ "$build_system" == "cmake" ]; then
+        if [[ ${compiler} == gnu ]]; then
+            export CC=gcc
+            export CXX=g++
+        elif [[ ${compiler} == clang ]]; then
+            export CC=clang
+            export CXX=clang++
+        elif [[ ${compiler} == icx ]]; then
+            export CC=icx
+            export CXX=icpx
+        fi
+        echo "============== Configuration: =============="
+        echo Compiler:  ${compiler}
+        echo Link mode: ${link_mode}
+        echo CC: ${CC}
+        echo CXX: ${CXX}
+        echo "============================================"
+
+        if [ -d "Build" ]; then
+            rm -rf ./Build/*
+        else
+            mkdir Build
+        fi
+
+        cmake -B./Build -S. -G "Unix Makefiles" -DTARGET_LINK=${link_mode} -DTBB_DIR=${TBBROOT}/lib/cmake/tbb
+        err=$?
+        if [ ${err} -ne 0 ]; then
+            echo -e "$(date +'%H:%M:%S') CMAKE GENERATE FAILED\t\t"
+            TESTING_RETURN=${err}
+            continue
+        fi
+        make -C Build
+        err=$?
+        if [ ${err} -ne 0 ]; then
+            echo -e "$(date +'%H:%M:%S') BUILD FAILED\t\t"
+            TESTING_RETURN=${err}
+            continue
+        fi
+        output_result=
+        err=
+        cmake_results_dir="_cmake_results/intel_intel64_${lib_ext}"
+        for p in ${cmake_results_dir}/*; do
+            e=$(basename "$p")
+            ${p} 2>&1 > ${e}.res
+            err=$?
+            output_result=$(cat ${e}.res)
+            mv -f ${e}.res ${cmake_results_dir}/
+            status_ex=
+            if [ ${err} -ne 0 ]; then
+                echo "${output_result}"
+                status_ex="$(date +'%H:%M:%S') FAILED\t\t${e} with errno ${err}"
+                TESTING_RETURN=${err}
+                continue
+            else
+                echo "${output_result}" | grep -i "error\|warn"
```

Comment:
Grep will output `OOB error = X` and other similar. Examples output or grep pattern should be rewritten

Reply:
yes. this is code and logic that we already use internally.  So i think better patterns should be aligned between both. 

Here it's better way to get things printed and not skipping them at all. Might be it's better to just skipp this output.